File: src/test/resources/collections/MapGetNullability.java
package collections;

import java.util.Map;

public class MapGetNullability {

    public Map<Long, String> stringMap;
    public Map<Long, Integer> intMap;

    public void work() {
        String str = stringMap.get(1L);
        if (str != null) {
            System.out.println(str);
        }

        Integer i = intMap.get(2L);
        if (i != null) {
            System.out.println(i);
            System.out.println(i + 1);
        }
        System.out.println(i);
        if (i == null || str == null) {
            System.out.println(i + "text");
            System.out.println("text" + i);
            System.out.println(i + str);
            System.out.println(str + i);
        }
        System.out.println(i + 2);

        print1(i, i);
        print2(i, i);
    }

    public void print1(int a, Object b) {
        System.out.println(a + " " + b);
    }

    public void print2(Integer a, int b) {
        System.out.println(a + b);
    }

}

File: src/test/resources/collections/MapGetNullability.cs
namespace collections {
    public class MapGetNullability {
        public System.Collections.Generic.IDictionary<long, string> stringMap;

        public System.Collections.Generic.IDictionary<long, int> intMap;

        public virtual void work() {
            string str = stringMap.GetOrNull(1L);
            if (str != null) {
                System.Console.Out.WriteLine(str);
            }
            int? i = intMap.GetOrNullable(2L);
            if (i != null) {
                System.Console.Out.WriteLine(i);
                System.Console.Out.WriteLine(i.Value + 1);
            }
            System.Console.Out.WriteLine(i);
            if (i == null || str == null) {
                System.Console.Out.WriteLine(i + "text");
                System.Console.Out.WriteLine("text" + i);
                System.Console.Out.WriteLine(i + str);
                System.Console.Out.WriteLine(str + i);
            }
            System.Console.Out.WriteLine(i.Value + 2);
            print1(i.Value, i);
            print2(i.Value, i.Value);
        }

        public virtual void print1(int a, object b) {
            System.Console.Out.WriteLine(a + " " + b);
        }

        public virtual void print2(int a, int b) {
            System.Console.Out.WriteLine(a + b);
        }
    }
}